tests/unit_tests/fn/test_extents.cpp

```diff
@@ -0,0 +1,99 @@
+/*
+ * GridTools
+ *
+ * Copyright (c) 2014-2021, ETH Zurich
+ * All rights reserved.
+ *
+ * Please, refer to the LICENSE file in the root directory.
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+#include <gridtools/fn/extents.hpp>
+
+#include <type_traits>
+
+#include <gtest/gtest.h>
+
+#include <gridtools/common/integral_constant.hpp>
+#include <gridtools/common/tuple_util.hpp>
+
+namespace gridtools::fn {
+    namespace {
+        template <class A,
+            class B,
+            class CommonValueType = std::common_type_t<typename A::value_type, typename B::value_type>>
+        struct integral_constant_equal
+            : std::is_same<integral_constant<CommonValueType, A::value>, integral_constant<CommonValueType, B::value>> {
+        };
+
+        using namespace literals;
+        struct a {};
+        struct b {};
+        struct c {};
+
+        static_assert(is_extent<extent<a, 0, 0>>::value);
+        static_assert(is_extent<extent<a, -1, 1>>::value);
+
+        namespace {
+            using testee = extents<extent<a, 0, 0>, extent<b, 0, 0>>;
+        }
+
+        namespace extents_offsets {
+            using testee = decltype(extents<extent<a, -1, 0>, extent<b, 0, 2>>::offsets());
+
+            static_assert(!has_key<testee, b>::value);
+            static_assert(integral_constant_equal<decltype(at_key<a>(std::declval<testee>())), decltype(-1_c)>::value);
+        } // namespace extents_offsets
+
+        namespace extents_sizes {
+            using testee = decltype(extents<extent<a, -1, 0>, extent<b, 0, 2>, extent<c, 1, 1>>::sizes());
+
+            static_assert(!has_key<testee, c>::value);
+            static_assert(integral_constant_equal<decltype(at_key<a>(std::declval<testee>())), decltype(1_c)>::value);
```

Comment:
you can use `element_at` from `hymap` here

Reply:
oh forgot that one, will update